app/src/main/java/de/tum/in/tumcampusapp/activities/FeedbackActivity.java

```diff
@@ -0,0 +1,363 @@
+            new File(path).delete();
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSION_LOCATION: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                    && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    location = new LocationManager(this).getLastLocation();
+                    includeLocation.setChecked(true);
+                } else {
+                    includeLocation.setChecked(false);
+                }
+                return;
+            }
+            case PERMISSION_CAMERA: {
+                if (grantResults.length > 0
+                    && grantResults[0] == PackageManager.PERMISSION_GRANTED){
+                    startTakingPicture();
+                }
+                return;
+            }
+            case PERMISSION_FILES: {
+                if (grantResults.length > 0
+                    && grantResults[0] == PackageManager.PERMISSION_GRANTED){
+                    openGallery();
+                }
+                return;
+            }
+        }
+    }
+
+    private boolean isValidEmail(){
+        String email = customEmailView.getText().toString();
+        boolean isValid = pattern.matcher(email).matches();
+        if(isValid){
+            customEmailView.setTextColor(getResources().getColor(R.color.valid));
+        } else {
+            customEmailView.setTextColor(getResources().getColor(R.color.error));
+        }
+        return isValid;
+    }
+
+    public void sendFeedback(View view){
+        if(includeEmail.isChecked() && "".equals(lrzId) && !isValidEmail()){
+            return;
+        }
+
+        // TODO send the feedback data to the server
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState){
+        super.onSaveInstanceState(outState);
+        outState.putInt(Const.FEEDBACK_TOPIC, feedbackTopic);
+        outState.putString(Const.FEEDBACK_MESSAGE, feedbackView.getText().toString());
+        outState.putStringArrayList(Const.FEEDBACK_PIC_PATHS, picPaths);
+        outState.putBoolean(Const.FEEDBACK_INCL_EMAIL, includeEmail.isChecked());
+        outState.putBoolean(Const.FEEDBACK_INCL_LOCATION, includeLocation.isChecked());
+        outState.putString(Const.FEEDBACK_EMAIL, customEmailView.getText().toString());
+    }
+
+    public void addPicture(View view){
+        AlertDialog.Builder dialog = new AlertDialog.Builder(this);
+        dialog.setTitle(R.string.feedback_add_picture);
+        dialog.setIcon(R.drawable.ic_add_photo_colored);
+        String[] options = {getString(R.string.feedback_take_picture), getString(R.string.gallery)};
+        dialog.setItems(options, (dialogInterface, i) -> {
+            if(i == 0){
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
[`@RequiresApi`](https://developer.android.com/reference/android/support/annotation/RequiresApi.html)

Reply:
I feel like it's more more readable the way it is (or what reason is there to do this?). Because if I put it into an extra method I'd have to surround the method call with if(Build.VERSION.SDK_INT ...) again

Reply:
The way it works with the support library is, you call the method unconditionally. This then feels like you always program against the latest SDK version.

The method then does the appropriate things it needs to do for each of the API versions